pkg/trace/writer/trace.go

```diff
@@ -212,27 +208,26 @@ outer:
 
 func (w *TraceWriter) resetBuffer() {
 	w.bufferedSize = 0
-	w.traces = w.traces[:0]
-	w.events = w.events[:0]
+	w.tracerPayloads = w.tracerPayloads[:0]
 }
 
 const headerLanguages = "X-Datadog-Reported-Languages"
 
 func (w *TraceWriter) flush() {
-	if len(w.traces) == 0 && len(w.events) == 0 {
+	if len(w.tracerPayloads) == 0 {
 		// nothing to do
 		return
 	}
 
 	defer timing.Since("datadog.trace_agent.trace_writer.encode_ms", time.Now())
 	defer w.resetBuffer()
 
-	log.Debugf("Serializing %d traces and %d APM events.", len(w.traces), len(w.events))
-	tracePayload := pb.TracePayload{
-		HostName:     w.hostname,
-		Env:          w.env,
-		Traces:       w.traces,
-		Transactions: w.events,
+	log.Debugf("Serializing %d tracer payloads.", len(w.tracerPayloads))
+	tracePayload := pb.AgentPayload{
```

Comment:
We should rename this variable to `p`, `agentPayload`, or something that makes sense now that we've renamed the type.

Reply:
☝🏻

Reply:
Also, how does this work? Does the same intake endpoint accept this new structure along with the old one? That seems odd. Isn't it needed to change the endpoint path to a new version?

Reply:
@gbbr On our side, we decided to use the same endpoint. If `TracerPayload` appears in the protobuf, we consider the payload is from the new agent.